db/log_writer.cc

```diff
@@ -154,7 +155,7 @@ IOStatus Writer::EmitPhysicalRecord(RecordType t, const char* ptr, size_t n) {
   // Write the header and the payload
   IOStatus s = dest_->Append(Slice(buf, header_size));
```

Comment:
Do we not have to pass the header checksum here?

Reply:
@anand1976 So you mean we can directly use crc32c::Value(buf, header_size) to calculate the checksum here and pass down to writable file writer, instead of wait Append calculate internally?

Reply:
@anand1976 If checksum handoff is not enabled, the calculated checksum of the header is useless. Therefore, it would be more efficient to call without the checksum and calculate the checksum internally when it is needed.